e2e/playwright/actions/src/tests/share/unshare-file-search-results.spec.ts

```diff
@@ -0,0 +1,211 @@
+      const docLibId = await sitesApi.getDocLibId(sitePrivate);
+
+      fileSite1Id = (await nodesApi.createFile(fileSite1, docLibId)).entry.id;
+      fileSite2Id = (await nodesApi.createFile(fileSite2, docLibId)).entry.id;
+
+      await sitesApi.addSiteMember(sitePrivate, username, Site.RoleEnum.SiteConsumer);
+
+      await shareApi.shareFilesByIds([fileSite1Id]);
+      await shareApi.waitForFilesToBeShared([fileSite1Id]);
+
+      await shareApi.shareFilesByIds([file1Id, file2Id, file3Id, file4Id, fileSite2Id]);
+      await shareApi.waitForFilesToBeShared([file1Id, file2Id, file3Id, file4Id, fileSite2Id]);
+
+      await filesAction.waitForNodes(`search-file-${searchRandom}`, { expect: 6 });
+    } catch (error) {
+      throw new Error(`----- beforeAll failed : ${error}`);
+    }
+  });
+
+  test.beforeEach(async ({ loginPage, personalFiles }) => {
+    await loginPage.navigate();
+    await loginPage.loginUser({ username: username, password: username });
+    await personalFiles.waitForPageLoad();
+  });
+
+  test.afterAll(async () => {
+    await apiClientFactory.nodes.deleteNode(parentId, { permanent: true });
+    await sitesApi.deleteSites([sitePrivate]);
+  });
+
+  test('[C306995] Unshare dialog UI', async ({ personalFiles, searchPage }) => {
+    await personalFiles.acaHeader.searchButton.click();
+    await searchPage.searchInput.searchButton.click();
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(file1);
+
+    await personalFiles.dataTable.selectItem(file1);
+    await personalFiles.acaHeader.shareButton.click();
+    await personalFiles.viewerDialog.shareDialogTitle.waitFor({ state: 'attached', timeout: timeouts.normal });
+
+    expect(await personalFiles.shareDialog.isShareToggleChecked()).toBe(true);
+    await personalFiles.shareDialog.shareToggle.click();
+
+    expect(personalFiles.confirmDialog.getDialogTitle('Remove this shared link')).toBeVisible();
+    expect(
+      personalFiles.confirmDialog.getDialogContent('This link will be deleted and a new link will be created next time this file is shared')
+    ).toBeVisible();
+    expect(personalFiles.confirmDialog.okButton).toBeEnabled();
+    expect(personalFiles.confirmDialog.cancelButton).toBeEnabled();
+  });
+
+  test('[C306996] Unshare a file', async ({ personalFiles, searchPage, nodesApiAction, page }) => {
+    await personalFiles.acaHeader.searchButton.click();
+    await searchPage.searchInput.searchButton.click();
+    await searchPage.searchOverlay.checkFilesAndFolders();
+    await searchPage.searchOverlay.searchFor(file2);
+
+    await personalFiles.dataTable.selectItem(file2);
+    await personalFiles.acaHeader.shareButton.click();
+    const url = await personalFiles.shareDialog.getLinkUrl();
+    await personalFiles.shareDialog.shareToggle.click();
+
+    await personalFiles.confirmDialog.okButton.click();
+    expect(await nodesApiAction.isFileShared(file2Id)).toBe(false);
```

Comment:
we can move this line after Check on UI as we are getting flaky test due to backend take time to reflect file status.